TheJumpCloud/support
# SA-2690: Private Function for Parallel Validation Check

## Issues
* [SA-2690](https://jumpcloud.atlassian.net/browse/SA-2690) - Private Function for Parallel Validation Check

## What does this solve?
The private function is called in place of manual checks to determine if a function can be ran in parallel or not. This reduces repetitive code

## Is there anything particularly tricky?
N/A

## How should this be tested?

* Attempt to run the following functions in Parallel (the -debug flag should inform you if the parallel validation passed/failed):
  * Get-JCUser
  * Get-JCSystem
  * Get-JCSystemUser
  * Get-JCSystemGroupMember
  * Get-JCUserGroupMember
  * Get-JCCommandResult
* If the version of PowerShell is greater than 7 and the $parallel parameter is set to $true, then the function will continue in parallel.
* If the version is less than 7, the version of PowerShell does not support parallel functionality and a warning will be displayed to the customer, the function will continue in sequential.
* If the $parallel parameter is set to $false, the function will continue in sequential
